### ISSUE TITLE
[Feature Request] mDNS support for Account URLs

### ISSUE THREAD

**Issue by StevenLares** [readrops/Readrops#193]
[This page explains mDNS](https://blog.matrixpost.net/what-is-multicast-dns-mdns-and-how-it-works/)

I have a server in my home network that's running FreshRSS. Let's say that the server's host name is **home-server**

I wish to add this FreshRSS instance to Readrop v2.0-beta-1 by going to Account > New Account > FreshRSS

Then, I enter **http://home-server:freshrss_port** under "Account URL" 

However, when I click "Validate", I get an error message:
```
Input is not a valid URL
```

Instead, I must use **http://local_IP:freshrss_port** or **http://home-server.local:freshrss_port**
Note the use of .local in the second option.

Some reasons why adding mDNS support would be useful:

1. No need to remember the server's local IP
2. Tailscale users can connect to their home servers by using the same mDNS hostnames available in their local network. This  extends to their FreshRSS instances.

In other words, if I can set Account URL to http://home-server:freshrss_port, then I can connect to FreshRSS while away from home (using Tailscale). 

When I get back home and directly connect to the network, I can continue to use the same RSS server seamlessly (without having to manually adjust the Account URL).

**Comment by Shinokuni** [comment]
Thank you @sevmonster for providing all those info. Indeed the most correct fix here seems to be fully removing the url validation with `android.util` Patterns. Yet some URL validations will still occur and will need to be addressed to the user. For example, Retrofit the REST client Readrops is using performs a minimal check but still, which can throw errors.

**Comment by sevmonster** [comment]
Honestly testing to see if the URL can connect would be the easiest metric, and would work for both URL validation and connection issues etc.

**Comment by Shinokuni** [comment]
So I made login checks less restrictive in 250c5c111d0bee9e795684790cee94b58624f8ee. I also tested with mDNS and it works, both with `.local` and without. It will be available in the next release.
